packages/abi-gen-templates/contract.handlebars

```diff
@@ -138,7 +144,88 @@ export class {{contractName}}Contract extends BaseContract {
             {{/each}}
         ] as ContractAbi;
         return abi;
+    }{{#if events}}
+    public async getSubscriptionManagerAsync(): Promise<SubscriptionManager<{{contractName}}EventArgs, {{contractName}}Events>> {
+        if (this._subscriptionManagerIfExists === undefined) {
+            const networkId = await this._web3Wrapper.getNetworkIdAsync();
+            this._subscriptionManagerIfExists = new SubscriptionManager<{{contractName}}EventArgs, {{contractName}}Events>(
+                {{contractName}}Contract.ABI(),
+                this._web3Wrapper,
+                networkId,
+            );
+        }
+        return this._subscriptionManagerIfExists;
+    }
+    /**
+     * Subscribe to an event type emitted by the {{contractName}} contract.
+     * @param   eventName           The {{contractName}} contract event you would like to subscribe to.
+     * @param   indexFilterValues   An object where the keys are indexed args returned by the event and
+     *                              the value is the value you are interested in. E.g `{maker: aUserAddressHex}`
+     * @param   callback            Callback that gets called when a log is added/removed
+     * @param   isVerbose           Enable verbose subscription warnings (e.g recoverable network issues encountered)
+     * @return Subscription token used later to unsubscribe
+     */
+    public async subscribeAsync<ArgsType extends {{contractName}}EventArgs>(
+        eventName: {{contractName}}Events,
+        indexFilterValues: IndexedFilterValues,
+        callback: EventCallback<ArgsType>,
+        isVerbose: boolean = false,
+    ): Promise<string> {
+        assert.doesBelongToStringEnum('eventName', eventName, {{contractName}}Events);
+        assert.doesConformToSchema('indexFilterValues', indexFilterValues, schemas.indexFilterValuesSchema);
+        assert.isFunction('callback', callback);
+        const subscriptionManager = await this.getSubscriptionManagerAsync();
```

Comment:
Let's make `getSubscriptionManagerAsync` into a private method.

Reply:
removing this one completely